resources/usa_mining/client/main.lua

```diff
@@ -83,9 +83,10 @@ end)
 
 Citizen.CreateThread(function()
     while true do
+        local player = PlayerPedId()
         if nearMarker(purchase_location.x, purchase_location.y, purchase_location.z, 3) then
             exports.globals:DrawText3D(purchase_location.x, purchase_location.y, purchase_location.z, '[E] - Sell Goods | [H] - Hint')
-            if IsControlJustPressed(0, 86) then
+            if IsControlJustPressed(0, 86) and not IsPedInAnyVehicle(player) then
```

Comment:
```suggestion
            if IsControlJustPressed(0, 86) and not IsPedInAnyVehicle(PlayerPedId()) then
```

only call it at minimum when necessary instead (like once they press the hotkey here)